src/library/recording/dlgrecording.cpp

```diff
@@ -120,7 +113,11 @@ void DlgRecording::setFocus() {
 }
 
 void DlgRecording::refreshBrowseModel() {
-    m_browseModel.setPath(mixxx::FileAccess(mixxx::FileInfo(m_recordingDir)));
+    saveCurrentViewState();
+    QString recordingDir = m_pRecordingManager->getRecordingDir();
+    // Hack: append "/" so the model stores the complete path the model state key
+    m_browseModel.setPath(mixxx::FileAccess(mixxx::FileInfo(
```

Comment:
Please note that neither mixxx::FileInfo nor QFileInfo promise to keep a trailing slashes.

It looks like the current mechanism for storing paths as model keys has serious shortcomings and these hacks may stop working at any time.

Reply:
The issue is slightly different anyway. See https://github.com/mixxxdj/mixxx/pull/4648#discussion_r796465379

Reply:
These checks is exactly what I wanted to avoid.

Back to square one:
Browse passes all paths with trailing / (manually appended, for no obvious reason).
Removing the / was actually just a cosmetic issue for the model key string and `path.info().locationPath()` a simple way to
* check if path "is valid" and
* remove /

Rec is the only feature that _was_ passing a fixed path _without_ trailing slash.
My change to append / there, too, is only working around `.locationPath()` chopping the last dir part.
We may simply check for `path.info().isDir()` and use the submitted `path` then.